volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,777 @@
+        if not mod_symbols:
+            return None
+
+        pdb_module = context.module(
+            mod_symbols, layer_name=layer_name, offset=module_start
+        )
+
+        return PDBSymbolFinder(layer_name, mod_name, module_start, pdb_module)
+
+    @staticmethod
+    def _find_symbols_through_pdb(
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        module_instances: List[Tuple[str, int, int]],
+        mod_name: str,
+    ) -> Generator[PDBSymbolFinder, None, None]:
+        """
+        Attempts to resolve the symbols in `wanted_symbols` through PDB analysis
+        """
+        for module_info in module_instances:
+            mod_module = PESymbols._get_pdb_module(
+                context, config_path, mod_name, module_info
+            )
+            if mod_module:
+                yield mod_module
+
+    @staticmethod
+    def _find_symbols_through_exports(
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        module_instances: List[Tuple[str, int, int]],
+        mod_name: str,
+    ) -> Generator[ExportSymbolFinder, None, None]:
+        """
+        Attempts to resolve the symbols in `wanted_symbols` through export analysis
+        """
+        pe_table_name = intermed.IntermediateSymbolTable.create(
+            context, config_path, "windows", "pe", class_types=pe.class_types
+        )
+
+        # for each process layer and VAD, construct a PE and examine the export table
+        for module_info in module_instances:
+            exported_symbols = PESymbols._get_exported_symbols(
+                context, pe_table_name, mod_name, module_info
+            )
+            if exported_symbols:
+                yield exported_symbols
+
+    @staticmethod
+    def _get_symbol_value(
+        wanted_modules: PESymbolFinder.cached_value_dict,
+        mod_name: str,
+        symbol_resolver: PESymbolFinder,
+    ) -> Generator[Tuple[str, int], None, None]:
+        """
+        Enumerates the symbols specified as wanted by the calling plugin
+        """
+        wanted_symbols = wanted_modules[mod_name]
+
+        if (
+            PESymbols.wanted_names not in wanted_symbols
+            and PESymbols.wanted_addresses not in wanted_symbols
+        ):
+            vollog.warning(
+                f"Invalid `wanted_symbols` sent to `find_symbols` for module {mod_name}. addresses and names keys both misssing."
+            )
+            return
+
+        symbol_keys = [
+            (PESymbols.wanted_names, "get_address_for_name"),
+            (PESymbols.wanted_addresses, "get_name_for_address"),
+        ]
+
+        for symbol_key, symbol_getter in symbol_keys:
+            # address or name
+            if symbol_key in wanted_symbols:
+                # walk each wanted address or name
+                for wanted_value in wanted_symbols[symbol_key]:
+                    symbol_value = symbol_resolver.__getattribute__(symbol_getter)(
```

Comment:
I don't really trust it?  `getattr` looks it up and returns it directly if the class provides it, or allows the `__getattr__` method to return it.  `__getattribute__` is calling a private member of the class and most type checkers should scream at doing that.

Since symbol_resolver doesn't change, and symbol_getter is only used one, why not change the string lookup table to a method lookup table, so:

```
conversion_method_lookup = [
(wanted_names_identifier, symbol_resolve.get_address_for_name),
(wanted_addresses_identifier, symbol_resolver.get_name_for_address)
]
```

Then later, you can just do `symbol_value = symbol_getter(wanted_value)` and it's a lot cleaner to read, understand and not have to worry about `getattr`?

Reply:
Yes, I will definitely change to this. I get what you mean now.